sql/src/test/java/io/crate/expression/scalar/ArrayUniqueFunctionTest.java

```diff
@@ -99,7 +99,7 @@ public void testOneArgument() throws Exception {
 
     @Test
     public void testDifferentButConvertableInnerTypes() throws Exception {
-        assertEvaluate("array_unique([10, 20], [10.1, 20.0])", Arrays.asList(10L, 20L));
```

Comment:
The rule that the first array dictates the array element type currently isn't supported by the new signature binder.
We also documented it here https://crate.io/docs/crate/reference/en/latest/general/builtins/scalar.html#array-unique-first-array-second-array, I didn't know. Currently the precedence dictates the common type and thus element type. I wonder why we've implemented it like that, any input here?

Reply:
I don't think there was a good reason for the behaviour. 

Not sure  if it's worth keeping the behaviour for BWC. Breaking changes entry?

Reply:
+1 for changing with a breaking change.
I've also checked the code history and found no argument to that behaviour. I think this was related to missing precedence support at the time when this was implemented.
Supporting it would add (quite) some extra complexity to the signature binder.